src/main/java/IslandBFS.java

```diff
@@ -0,0 +1,57 @@
+import java.util.*;
+
+public class IslandBFS {
+
+    int[][] multi;
+    Queue<Point> queue = new LinkedList<Point>();
+    Set<Point> lands = new HashSet<>();
+    int numOfIslands = 0;
+
+
+    public IslandBFS(int[][] multi) {
+        this.multi = multi;
+    }
+
+    public int scanMap() {
+        Point point = new Point(0, 0);
+        for (int i = 0; i < multi.length; i++) {
+            for (int j = 0; j < multi[0].length; j++) {
+                point.r = i;
+                point.c = j;
+                if (multi[i][j] == 1 && !lands.contains(point)) {
+                    lands.add(point);
```

Comment:
Make a function to expand a point here and put all the following code inside it so that it is more readable.